### PR TITLE
Task#55720:[Refresh link]create a scan broken bookmark tool.

### PR DESCRIPTION
**Pull request by v-haiboz**

### CODE REVIEW

File: Managers/BookmarkManager.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                {
+                    try
+                    {
+                        var bookmarkurl = await GetProblematicBookmark(filePath);
+                        if (bookmarkurl != null && bookmarkurl.Count > 0)
+                        {
+                            Add(new FilterResult()
+                            {
+                                MarkdownPath = filePath,
+                                Bookmarkurl = await GetProblematicBookmark(filePath)
```
Comment (yunair) [inline Managers/BookmarkManager.cs]: why we need to rerun `await GetProblematicBookmark(filePath)`? can we use `bookmarkurl` directly?

Comment (v-haiboz) [reply Managers/BookmarkManager.cs]: Maybe some bookmarks are right in the file. Here we should filter out the right ones

File: Analyzers/AnchorPointAnalyzer.Extractor.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                }
+
+                match = match.NextMatch();
+            }
+        }
+
+        private async static Task<string> GetAnchorPoint(string str)
+        {
+
+            if (string.IsNullOrEmpty(str) || !str.StartsWith("/"))
```
Comment (yunair) [inline Analyzers/AnchorPointAnalyzer.Extractor.cs:37]: if  str.StartsWith("https://docs.microsoft.com"),  will return string.empty

File: Analyzers/AnchorPointAnalyzer.Extractor.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                return string.Empty;
+            }
+
+            if (str.StartsWith("https://docs.microsoft.com"))
```
Comment (yunair) [inline Analyzers/AnchorPointAnalyzer.Extractor.cs:42]: if not found #, we need to return string.Empty

Comment (v-haiboz) [reply Analyzers/AnchorPointAnalyzer.Extractor.cs:42]: it has been handled.

var startIndex = str.IndexOf("#");
            if (startIndex == -1 || startIndex == str.Length)
            {
                return string.Empty;
            }

File: Analyzers/AnchorPointAnalyzer.Extractor.cs (hunk 2 above)
Comment (v-haiboz) [reply Analyzers/AnchorPointAnalyzer.Extractor.cs]: fixed

File: Analyzers/AnchorPointAnalyzer.Extractor.cs (hunk 3 above)
Comment (yunair) [reply Analyzers/AnchorPointAnalyzer.Extractor.cs:42]: we need return string.Empty before check str.StartsWith("https://docs.microsoft.com"),
if if (str.StartsWith("https://docs.microsoft.com")) but the str does not have #, we don't concern such url

Comment (v-haiboz) [reply Analyzers/AnchorPointAnalyzer.Extractor.cs]: Updated

**Review by yunair — APPROVED** [review]

**Pull request merged**